extensions/gmsa-dc/v1/Setup-gMSA.ps1

```diff
@@ -0,0 +1,271 @@
+                return
+            }
+        }
+    }
+
+    # Get the location to store the cred spec file either from input params or helper function
+    if ($Path) {
+        $CredSpecRoot = Split-Path $Path -Parent
+        $FileName = Split-Path $Path -Leaf
+    } else {
+        $CredSpecRoot = "C:\ProgramData\docker\credentialspecs"
+    }
+
+    if (-not $FileName) {
+        $FileName = "{0}_{1}" -f $ADDomain.NetBIOSName.ToLower(), $AccountName.ToLower()
+    }
+
+    $FullPath = Join-Path $CredSpecRoot "$($FileName.TrimEnd(".json")).json"
+    if ((Test-Path $FullPath) -and $NoClobber) {
+        Write-Error "A credential spec already exists with the name `"$FileName`".`nRemove the -NoClobber switch to overwrite this file or select a different name using the -FileName parameter."
+        return
+    }
+
+    # Start hash table for output
+    $output = @{}
+
+    # Create ActiveDirectoryConfig Object
+    $output.ActiveDirectoryConfig = @{}
+    $output.ActiveDirectoryConfig.GroupManagedServiceAccounts = @( @{"Name" = $AccountName; "Scope" = $ADDomain.DNSRoot } )
+    $output.ActiveDirectoryConfig.GroupManagedServiceAccounts += @{"Name" = $AccountName; "Scope" = $ADDomain.NetBIOSName }
+    if ($AdditionalAccounts) {
+        $AdditionalAccounts | ForEach-Object {
+            $output.ActiveDirectoryConfig.GroupManagedServiceAccounts += @{"Name" = $_.AccountName; "Scope" = $_.Domain }
+        }
+    }
+    
+    # Create CmsPlugins Object
+    $output.CmsPlugins = @("ActiveDirectory")
+
+    # Create DomainJoinConfig Object
+    $output.DomainJoinConfig = @{}
+    $output.DomainJoinConfig.DnsName = $ADDomain.DNSRoot
+    $output.DomainJoinConfig.Guid = $ADDomain.ObjectGUID
+    $output.DomainJoinConfig.DnsTreeName = $ADDomain.Forest
+    $output.DomainJoinConfig.NetBiosName = $ADDomain.NetBIOSName
+    $output.DomainJoinConfig.Sid = $ADDomain.DomainSID.Value
+    $output.DomainJoinConfig.MachineAccountName = $AccountName
+
+    $output | ConvertTo-Json -Depth 5 | Out-File -FilePath $FullPath -Encoding ascii -NoClobber:$NoClobber
+	
+	Install-Module powershell-yaml -Force
+
+    $CredSpecJson = Get-Item $FullPath | Select-Object @{
+        Name       = 'Name'
+        Expression = { $_.Name }
+    },
+    @{
+        Name       = 'Path'
+        Expression = { $_.FullName }
+    }
+	
+	# This section of code borrowed from https://github.com/kubernetes-sigs/windows-gmsa/blob/master/scripts/GenerateCredentialSpecResource.ps1
+    $dockerCredSpecPath = $CredSpecJson.Path
+    Sleep 2
+    $credSpecContents = Get-Content $dockerCredSpecPath | ConvertFrom-Json
+    $ManifestFile = "gmsa-cred-spec-gmsa-e2e.yml"
+	
+	# generate the k8s resource
+    $resource = [ordered]@{
+        "apiVersion" = "windows.k8s.io/v1alpha1";
+        "kind" = 'GMSACredentialSpec';
+        "metadata" = @{
+        "name" = $AccountName
+    };
+    "credspec" = $credSpecContents
+}
+
+ConvertTo-Yaml $resource | Set-Content $ManifestFile
```

Comment:
nit: the indentation here looks off. Please check the indentation. Should be fixed as part of fixing the indentation for the `}`s above.

Reply:
sorry, I may have mixed \t and `    ` at the source